kvclientmanager.go

```diff
@@ -252,6 +253,27 @@ func (m *kvClientManager) GetClient(ctx context.Context, endpoint string) (KvCli
 
 	return connProvider.GetClient(ctx)
 }
+func (m *kvClientManager) Close() error {
+	state, err := m.getState()
+	if err != nil {
+		return err
+	}
+
+	m.logger.Info("closing kv client manager")
+
+	m.state.Store(nil)
```

Comment:
There is a race here between m.getState() and m.state.Store(nil).  I think this would need to be a CompareAndSwap, or it should really be taking the same lock we use during reconfigure to avoid multiple concurrent calls from stepping on each others toes.